lib/registration.pm

```diff
@@ -96,20 +96,35 @@ sub fill_in_registration_data {
             send_key "alt-a";              # accept license
             send_key "alt-n";              # next
         }
-        for $a (split(/,/, get_var('SCC_ADDONS'))) {
-            $a = uc $a;                    # change to uppercase to match variable
-            if (my $regcode = get_var("SCC_REGCODE_$a")) {
-                assert_screen("scc-addon-regcode-$a");
-                send_key 'tab';            # jump to code field
-                type_string $regcode;
-                send_key "alt-n";          # next
+        if (check_screen("scc-addon-regcodes", 5)) {    # no step of input regcodes and skip it if register via smt
+            for $a (split(/,/, get_var('SCC_ADDONS'))) {
+                next if ($a !~ /ha|geo|we/);   # skip addons that no need regcode
```

Comment:
I just simply add a if statement out of inputing regcodes snippet to avoid input regcodes when registering via smt